bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartWebSocket.java

```diff
@@ -175,7 +188,7 @@ private void sendKeepAlivePing() {
                 session.close(1000, "Timeout manually closing dead connection (" + socketId + ")");
```

Comment:
There are two errors here..
1. Don't send a ping to the server if we are going to immediately close it anyway.
2. Don't (try to) ping the server if the websocket is not open.

Suggested code is as follows..

```
   private void sendKeepAlivePing() {
        final PostOAuth2Response accessToken = token;
        if ((Instant.now().getEpochSecond() - lastPong.getEpochSecond() > WEBSOCKET_IDLE_TIMEOUT) || accessToken == null
                || accessToken.isAccessTokenExpired()) {
            session.close(1000, "Timeout manually closing dead connection (" + socketId + ")");
        } else {
            if (session.isOpen()) {
                try {
                    logger.trace("Sending ping ({})", socketId);
                    session.getRemote().sendPing(pingPayload);
                } catch (IOException ex) {
                    logger.debug("Error while sending ping: {}", ex.getMessage());
                }
            }
        }
    }
```